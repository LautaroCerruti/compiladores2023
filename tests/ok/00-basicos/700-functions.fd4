let rec suma (m n:Nat) : Nat = 
    ifz m then n else suma (m-1) (n+1)

let sumTest : Nat = print "suma 150 + 163 = " (suma 150 163)

let rec mult (m n:Nat) : Nat = 
    ifz m then 0 else suma n (mult (m-1) n)

let multTest : Nat = print "multiplicacion 55 * 37 = " (mult 55 37)

let rec fact (n:Nat) : Nat = 
    ifz n then 1 else mult n (fact (n-1))

let factTest : Nat = print "factorial de 7 = " (fact 7)

let rec fib (n:Nat) : Nat = 
    ifz n then 0 else (
        ifz (n-1) then 1 else suma (fib (n-1)) (fib (n-2))
    )

let fibTest : Nat = print "fibonacci de 10 = " (fib 10)