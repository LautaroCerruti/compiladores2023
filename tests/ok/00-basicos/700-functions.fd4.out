suma 150 + 163 = 313
multiplicacion 55 * 37 = 2035
factorial de 7 = 5040
fibonacci de 10 = 55
